### ISSUE TITLE
Exploratory Analysis 

### ISSUE THREAD

**Issue by JoeyZwicker** [pachyderm/pachyderm#728]
Capturing an offline discussion. We'd like to make exploratory data analysis smoother in Pachyderm. Even with Update Pipeline and jobs, fast iteration is a pretty kludgy process -- write some code, build a container, and update the JSON pipeline spec. 

One idea for making this better is something like Pachctl Shell, `pachctl sh`. Basically, a REPL to interactively  explore data in repo(s) and then an smooth process for capturing that and "productionizing" it into a pipeline. Tools like iPython or SQL notebooks might be a good interface to support/mimic.

**Comment by ysimonson** [comment]
@JoeyZwicker this is done now, no?

**Comment by JoeyZwicker** [comment]
No, this is not done. I renamed this issue because the Pachctl Shell (which does something pretty different) is feature we just released and changing to REPL is an attempt to differentiate. #3709 has additional information of the use case for this.
